WarOOP/Models/Battle.cs

```diff
@@ -0,0 +1,62 @@
+using System.Diagnostics;
+
+namespace WarOOP.Models;
+
+public static class Battle
+{
+    public static bool Fight(Warrior warrior1, Warrior warrior2)
+    {
+        while (true)
+        {
+            warrior1.AttackTo(warrior2);
+            if (!warrior2.IsAlive)
+            {
+                return true;
+            }
+
+            warrior2.AttackTo(warrior1);
+            if (!warrior1.IsAlive)
+            {
+                return false;
+            }
+        }
+    }
+
+    public static bool Fight(Army army1, Army army2)
+    {
+        if (army1 == null || army2 == null)
+        {
+            throw new NullReferenceException("Army can not be null");
+        }
+        
+        if (!army1.HasUnits() && !army2.HasUnits())
+        {
+            throw new Exception("Armies can not be empty");
+        }
+
+        if (!army1.HasUnits())
```

Comment:
I see duplicated checks for empty armies

Reply:
This is necessary for the automatic victory of one of the armies if the opposite is empty